### ISSUE TITLE
react-hot-loader deprecated

### ISSUE THREAD

**Issue by brownieboy** [survivejs/webpack_react#248]
Another day, another build tool change...

As per https://github.com/gaearon/react-hot-loader, the react-hot-loader project has now been deprecated in favour of the same developer's "React Transform family of projects", see https://github.com/gaearon/react-transform-boilerplate

Are there plans to update chapter 3.4 etc to use this new tool?

FYI, I have it working with webpack-dev-server, but am unable to exclude it from my production build process.  So when I run my built files in the browser, I get the console error "locals[0] does not appear to be a `module` object with Hot Module replacement API enabled".  It seems that I'm not the only one with this problem: https://github.com/gaearon/react-transform-hmr/issues/5

This new module complicates matters, IMHO, by pushing a lot of config to the .babelrc file.  You're supposed to be able to pass the NODE_ENV variable to babel to set which options are included for production vs development, but I haven't managed to get it working.

**Comment by bebraw** [comment]
Yeah, I'm aware of the deprecation. It came quicker than I anticipated. Before committing to a solution, I've taken an observational stance. I don't want to tweak the configuration too many times as that gets painful for everyone. :+1:

I've experimented with the new style configuration at [react-demos](https://github.com/survivejs/react-demos). I've pushed the Babel bits to `.babelrc` there. I can see @bkniffler has pushed the development bits to [Webpack configuration](https://github.com/gaearon/react-transform-hmr/issues/5#issuecomment-142429722). I can see the merit of that but I'm not sure if that's ideal. 

What if you want to enable some Babel features specifically for both prod/dev while wanting to maintain an external `.babelrc` for production config (needed for `webpack.config.babel.js` scheme to work)?

Furthermore, I noticed that even though the new solution can provide nice traces within the browser, this relies on `devtool: 'eval'`. Of course in this case the quality of the source maps isn't there and it gets harder to debug through Chrome Inspector... If you use higher quality source maps, Chrome Inspector looks fine but you lose nice traces given they point at the source bundle (not enough info for it to pull exact lines I think).

I would love to find nice, stable solution for this that allows us to get good source maps while keeping the configuration sane. We probably can't avoid bringing a heap of new dependencies to the project given it's so modular but there's not much we can do about that.
